cassdegrees/ui/views/lists.py

```diff
@@ -0,0 +1,107 @@
+from django.contrib.auth.decorators import login_required
+
+from api.models import CourseModel, SubplanModel, ProgramModel, ListModel
+from api.views import search
+
+from django.http import HttpResponseNotFound, HttpRequest
+from django.shortcuts import render, redirect
+from django.utils import timezone
+
+from ui.forms import EditListFormSnippet
+
+admin_url_prefix = "/staff/"
+list_program_url = admin_url_prefix + "list/?view=Program"
+
+list_course_url = admin_url_prefix + "list/?view=Course"
+
+
+@login_required
+def create_list(request):
+    duplicate = request.GET.get('duplicate', 'false')
+    if duplicate == 'true':
+        duplicate = True
+    elif duplicate == 'false':
+        duplicate = False
+
+    # Initialise instance with an empty string so that we don't get a "may be referenced before assignment" error below
+    instance = ""
+
+    # If we are creating a list from a duplicate, we retrieve the instance with the given id
+    # (should always come along with 'duplicate' variable) and return that data to the user.
+    if duplicate:
+        id = request.GET.get('id')
+        if not id:
+            return HttpResponseNotFound("Specified ID not found")
+        # Find the list to specifically create from:
+        instance = ListModel.objects.get(id=int(id))
+
+    if request.method == 'POST':
+        form = EditListFormSnippet(request.POST)
+
+        # todo: implement list view on admin view page then change url
+        if form.is_valid():
+            form.save()
+            return redirect(list_course_url + '&msg=Successfully Added List!')
+
+    else:
+        if duplicate:
+            form = EditListFormSnippet(instance=instance)
+        else:
+            form = EditListFormSnippet()
+
+    return render(request, 'createlist.html', context={
+        "edit": False,
+        "form": form,
+    })
+
+
+# Todo: implement list deletion with associated business rules (if any)
```

Comment:
Issue for this one as well if possible after merge

Reply:
Issue created